src/Broadway/Bundle/BroadwayBundle/DependencyInjection/BroadwayExtension.php

```diff
@@ -70,7 +70,27 @@ private function loadSagaStateRepository(array $config, ContainerBuilder $contai
                     'broadway.saga.state.mongodb_repository'
                 );
 
-                $container->setParameter('broadway.saga.mongodb.storage_suffix', $config['mongodb']['storage_suffix']);
+                $database = 'broadway_%kernel.environment%%broadway.saga.mongodb.storage_suffix%';
+
+                if (isset($config['mongodb']['connection'])) {
+
+                    if (isset($config['mongodb']['connection']['database'])) {
+                        $database = $config['mongodb']['connection']['database'];
+                    }
+
+                    $mongoConnection = $container->getDefinition('broadway.saga.state.mongodb_connection');
+
+                    if (isset($config['mongodb']['connection']['dsn'])) {
+                        $mongoConnection->replaceArgument(0, $config['mongodb']['connection']['dsn']);
```

Comment:
Nope, sadly not. Tried just collecting and setting the arguments, but that didn't work well, as both are optional...